klayout/drc/testing/run_regression.py

```diff
@@ -0,0 +1,924 @@
+    orignal_testcase : string or Path object
+        Path string to the orignal testcase
+
+    marker_testcase : string or Path
+        Path of the output marker gds file generated from db file.
+
+    Returns
+    -------
+    merged_gds_path : string or Path
+        Path of the final merged gds file generated.
+    """
+
+    new_lib = gdstk.Library()
+
+    lib_org = gdstk.read_gds(orignal_testcase)
+    lib_marker = gdstk.read_gds(marker_testcase)
+
+    # Getting flattened top cells
+    top_cell_org = lib_org.top_level()[0].flatten(apply_repetitions=True)
+    top_cell_marker = lib_marker.top_level()[0].flatten(apply_repetitions=True)
+    marker_polygons = top_cell_marker.get_polygons(apply_repetitions=True, include_paths=True, depth=None)
+
+    # Merging all polygons of markers with original testcase
+    for marker_polygon in marker_polygons:
+        top_cell_org.add(marker_polygon)
+
+    # Adding flattened merged cell
+    new_lib.add(top_cell_org.flatten(apply_repetitions=True))
+
+    # Writing final merged gds file
+    merged_gds_path = f'{marker_testcase.replace(".gds", "")}_merged.gds'
+    new_lib.write_gds(merged_gds_path)
+
+    return merged_gds_path
+
+
+def draw_polygons(polygon_data, cell, lay_num, lay_dt, path_width):
+    """
+    This function is used for drawing gds file with all violated polygons.
+
+    Parameters
+    ----------
+    polygon_data : str
+        Contains data points for each violated polygon
+    cell: gdstk.Cell
+        Top cell will contains all generated polygons
+    lay_num: int
+        Number of layer used to draw violated polygons
+    lay_dt : int
+        Data type of layer used to draw violated polygons
+    path_width : float
+        Width  will used to draw edges
+
+    Returns
+    -------
+    None
+    """
+
+    # Cleaning data points
+    polygon_data = re.sub(r'\s+', '', polygon_data)
+    polygon_data = re.sub(r'[()]', '', polygon_data)
+
+    tag_split = polygon_data.split(":")
+    tag = tag_split[0]
+    poly_txt = tag_split[1]
+    polygons = re.split(r"[/|]", poly_txt)
+
+    # Select shape type to be drawn
+    if tag == "polygon":
+        for poly in polygons:
+            points = [(float(p.split(",")[0]), float(p.split(",")[1])) for p in poly.split(";")]
+            cell.add(gdstk.Polygon(points, lay_num, lay_dt))
+
+    elif tag == "edge-pair":
+        for poly in polygons:
+            points = [(float(p.split(",")[0]), float(p.split(",")[1])) for p in poly.split(";")]
+            cell.add(gdstk.FlexPath(points, path_width, layer=lay_num, datatype=lay_dt))
+
+    elif tag == "edge":
+        for poly in polygons:
+            points = [(float(p.split(",")[0]), float(p.split(",")[1])) for p in poly.split(";")]
+            cell.add(gdstk.FlexPath(points, path_width, layer=lay_num, datatype=lay_dt))
+    else:
+        logging.error(f"## Unknown type: {tag} ignored")
+
+
+def convert_results_db_to_gds(results_database: str, rules_tested: list):
```

Comment:
curious if the same could be achieved with `klayout.db`?

Reply:
Same as above

Reply:
Let's consolidate this discussion with the other comment.